standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -648,260 +400,85 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
-            ps.setString(1, state);
-            ps.setString(2, rqst.getUser());
-            ps.setString(3, rqst.getHostname());
-            ps.setInt(4, txnType.getValue());
-            ps.addBatch();
-
-            if ((i + 1) % maxBatchSize == 0) {
-              txnIds.addAll(executeTxnInsertBatchAndExtractGeneratedKeys(dbConn, genKeySupport, ps, true));
-            }
-          }
-          if (numTxns % maxBatchSize != 0) {
-            txnIds.addAll(executeTxnInsertBatchAndExtractGeneratedKeys(dbConn, genKeySupport, ps, true));
-          }
-        }
-      }
-
-      assert txnIds.size() == numTxns;
-
-      addTxnToMinHistoryLevel(dbConn, txnIds, minOpenTxnId);
-
-      if (isReplayedReplTxn) {
-        List<String> rowsRepl = new ArrayList<>(numTxns);
-        List<String> params = Collections.singletonList(rqst.getReplPolicy());
-        List<List<String>> paramsList = new ArrayList<>(numTxns);
-        for (int i = 0; i < numTxns; i++) {
-          rowsRepl.add("?," + rqst.getReplSrcTxnIds().get(i) + "," + txnIds.get(i));
-          paramsList.add(params);
-        }
-
-        insertPreparedStmts = sqlGenerator.createInsertValuesPreparedStmt(dbConn,
-                "\"REPL_TXN_MAP\" (\"RTM_REPL_POLICY\", \"RTM_SRC_TXN_ID\", \"RTM_TARGET_TXN_ID\")", rowsRepl,
-                paramsList);
-        for (PreparedStatement pst : insertPreparedStmts) {
-          pst.execute();
-        }
-      }
-
-      if (transactionalListeners != null && !isHiveReplTxn) {
-        MetaStoreListenerNotifier.notifyEventWithDirectSql(transactionalListeners,
-                EventMessage.EventType.OPEN_TXN, new OpenTxnEvent(txnIds, txnType), dbConn, sqlGenerator);
-      }
-      return txnIds;
-    } finally {
-      if (insertPreparedStmts != null) {
-        for (PreparedStatement pst : insertPreparedStmts) {
-          pst.close();
-        }
-      }
-    }
-  }
+      LOG.error("OpenTxnTimeOut exceeded commit duration {}, deleting transactionIds: {}", elapsedMillis, txnIds);
 
-  private List<Long> executeTxnInsertBatchAndExtractGeneratedKeys(Connection dbConn, boolean genKeySupport,
-      PreparedStatement ps, boolean batch) throws SQLException {
-    List<Long> txnIds = new ArrayList<>();
-    if (batch) {
-      ps.executeBatch();
-    } else {
-      // For slight performance advantage we do not use the executeBatch, when we only have one row
-      ps.execute();
-    }
-    if (genKeySupport) {
-      try (ResultSet generatedKeys = ps.getGeneratedKeys()) {
-        while (generatedKeys.next()) {
-          txnIds.add(generatedKeys.getLong(1));
-        }
-      }
-    } else {
-      try (PreparedStatement pstmt =
-          dbConn.prepareStatement("SELECT \"TXN_ID\" FROM \"TXNS\" WHERE \"TXN_STATE\" = ?")) {
-        pstmt.setString(1, TXN_TMP_STATE);
-        try (ResultSet rs = pstmt.executeQuery()) {
-          while (rs.next()) {
-            txnIds.add(rs.getLong(1));
-          }
+      if (!txnIds.isEmpty()) {
+        try {
```

Comment:
why not to keep it in deleteInvalidOpenTransactions() private method, better readability

Reply:
done